src/simd/distances_avx.cc

```diff
@@ -38,14 +40,38 @@ masked_read(int d, const float* x) {
     // cannot use AVX2 _mm_mask_set1_epi32
 }
 
+namespace {
+
+inline __m256
+avx2_bf16_patch(__m256&& f256) {
+    if (!enable_avx_patch_fp32_bf16) {
+        return f256;
+    } else {
+        __m256i mask = _mm256_set1_epi32(0xFFFF0000);
```

Comment:
Seems not compatible with sealed index implementation. It does not just truncate fp32 value to bf16.

Reply:
good catch!